### PR TITLE
CI: Use python38 on el8 with ansible-core 2.12 and python39 on el9 with ansible-core 2.13 

### PR DESCRIPTION
**Pull request by mnecas**
Fixes: https://github.com/oVirt/ovirt-ansible-collection/issues/491

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -27,9 +27,20 @@ jobs:
       - name: Update pip
         run: pip3 install pip --upgrade
```
Comment (mwperina) [inline .github/workflows/build.yaml:28]: So this is relevant ony for CS9, right?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -27,9 +27,20 @@ jobs:
...
 
-      - name: Install pip modules
+      - name: Install python3.8 modules
+        if: ${{ matrix.shortcut == 'cs8' }}
         run: >
-          pip3 install pycodestyle pylint==2.4.4 voluptuous yamllint "rstcheck<3.5.0" antsibull-changelog "rich<11.0.0" "ansible-lint<5.0.0" 'cryptography<37.0.0'
+          pip-3.8 install pycodestyle pylint==2.4.4 voluptuous yamllint "rstcheck<3.5.0" antsibull-changelog "rich<11.0.0" "ansible-lint<5.0.0" 'cryptography<37.0.0'
```
Comment (mwperina) [inline .github/workflows/build.yaml:33]: Also once we moved to Python 3.8, do we really need those version limitation around cryptography, rstcheck, ...?

**Review by mwperina — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 2 above)
Comment (mnecas) [reply .github/workflows/build.yaml:33]: good point, will remove it

File: .github/workflows/build.yaml (hunk 1 above)
Comment (mnecas) [reply .github/workflows/build.yaml:28]: I still think so, I know there were some pip installation problems without it.
I'll create one testing commit without CS9

Comment (mnecas) [reply .github/workflows/build.yaml]: Looks like it was related only to python36, both passed the installation even without the upgrade.
Removing

**Comment by nirs** [comment]
So now we need to build ovirt-imageio for python 3.9 in the centos 9 build?

**Comment by mnecas** [comment]
> So now we need to build ovirt-imageio for python 3.9 in the centos 9 build?

the default python on centos 9 stream is python 3.9, so if you are building the ovirt-imageio for the el9 should be fine.
You don't need to do that python versioning magic as on centos 8 stream.

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**